LoopUp.Siesta/ServiceCollectionExtensions/AddSiestaClientExtensions.cs

```diff
@@ -0,0 +1,71 @@
+namespace LoopUp.Siesta.ServiceCollectionExtensions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http.Headers;
+    using LoopUp.Siesta.HttpDelegatingHandlers;
+    using Microsoft.Extensions.DependencyInjection;
+    using Serilog;
+
+    /// <summary>
+    /// Extension methods for <see cref="IServiceCollection"/> that can be used to add a Siesta client.
+    /// </summary>
+    public static class AddSiestaClientExtensions
+    {
+        /// <summary>
+        /// Add a Siesta Client of type T. This will include the use of Serilog logging and a correlationId.
+        /// </summary>
+        /// <param name="services">The IServiceCollection this.</param>
+        /// <param name="baseAddress">Base address Uri for the Siesta client.</param>
+        /// <param name="systemName">The name of the calling system.</param>
+        /// <param name="defaultHeaders">Any default headers you which to apply to the client.</param>
+        /// <param name="authenticationHeaderValue">Optional value for the authentication header.</param>
+        /// <param name="loggerCorrelationIdKey">Optional key for the logging correlation Id.</param>
+        /// <param name="requestHeaderCorrelationIdKey">Optional key for where to find the correlation id in the request.</param>
+        /// <typeparam name="T">The type of Siesta Client to add.</typeparam>
+        /// <returns>The IServiceCollection.</returns>
+        public static IServiceCollection AddSiestaClientWithCorrelationIdAndSerilog<T>(
```

Comment:
This method name feels a bit clunky - if we add another parameter it's going to be a hell of a mouthful. Can we not take `SiestaClientConfigurationOptions` as a parameter rather than the individual parameters?

Reply:
I honestly thought I had already made this change, but clearly I have not. FYI all changes will appear in the "whole lot" PR